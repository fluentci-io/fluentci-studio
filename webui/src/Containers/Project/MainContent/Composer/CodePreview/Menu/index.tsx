import Menu from "./MenuWithData";

export default Menu;
