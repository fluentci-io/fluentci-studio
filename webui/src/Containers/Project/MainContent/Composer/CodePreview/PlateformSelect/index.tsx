import PlateformSelect from "./PlateformSelectWithData";

export default PlateformSelect;
