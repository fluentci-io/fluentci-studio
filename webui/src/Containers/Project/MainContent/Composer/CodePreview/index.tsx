import CodePreview from "./CodePreviewWithData";

export default CodePreview;
