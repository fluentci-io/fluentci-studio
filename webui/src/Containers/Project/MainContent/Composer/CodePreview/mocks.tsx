export const code = `name: lint and test
on:
  push:
    branches:
      - main
jobs:
  tests:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - name: Setup Fluent CI
        uses: fluentci-io/setup-fluentci@v5
        with:
          wasm: true
          pipeline: deno
          args: |
            fmt
            lint
            test
            coverage
      - name: Upload to Codecov
        run: fluentci run --wasm codecov upload
        env:
          CODECOV_TOKEN: \${{ secrets.CODECOV_TOKEN }}`;
