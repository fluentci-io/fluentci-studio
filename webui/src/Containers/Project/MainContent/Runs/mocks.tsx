export const runs = [
  {
    id: "edggjsrj42ysd3gljsksn4tc",
    branch: "main",
    commit: "64326a0",
    date: "2024-05-30T14:22:57.031Z",
    project: "admiring-meninsky",
    projectId: "ccogwb1oau3y9oav8zwbg9zk",
    duration: 41185,
    jobs: [
      {
        id: "ycsbwbbi0rvlh5mnsjb7lp0p",
        name: "test",
        createdAt: "2024-05-30T14:22:57.017Z",
        status: "SUCCESS",
        duration: 23847,
        __typename: "Job",
      },
      {
        id: "maq4cxhvyirvvzx19xxwzbe5",
        name: "build",
        createdAt: "2024-05-30T14:22:57.017Z",
        status: "SUCCESS",
        duration: 16992,
        __typename: "Job",
      },
    ],
    message: null,
    name: "Run #20",
    title: "[plugin] update version",
    cursor: "IcHZlXgiwAAA",
    status: "SUCCESS",
    __typename: "Run",
  },
  {
    id: "mwf27469282wpoilujzzsfag",
    branch: "main",
    commit: "64326a0",
    date: "2024-05-30T14:20:54.866Z",
    project: "admiring-meninsky",
    projectId: "ccogwb1oau3y9oav8zwbg9zk",
    duration: 68358,
    jobs: [
      {
        id: "hrf5wd9agnkic1hapd6r94vp",
        name: "test",
        createdAt: "2024-05-30T14:20:54.860Z",
        status: "SUCCESS",
        duration: 33102,
        __typename: "Job",
      },
      {
        id: "if8ntcrxe71sqfbf2v8oojhk",
        name: "build",
        createdAt: "2024-05-30T14:20:54.860Z",
        status: "SUCCESS",
        duration: 34942,
        __typename: "Job",
      },
    ],
    message: null,
    name: "Run #19",
    title: "[plugin] update version",
    cursor: "IcHZlXgiwAAA",
    status: "SUCCESS",
    __typename: "Run",
  },
  {
    id: "q0xrwtha16ayq2c3z28dvc9i",
    branch: "main",
    commit: "64326a0",
    date: "2024-05-30T14:19:17.866Z",
    project: "admiring-meninsky",
    projectId: "ccogwb1oau3y9oav8zwbg9zk",
    duration: 39889,
    jobs: [
      {
        id: "v83f464e4rn42o90m2uh8pa3",
        name: "test",
        createdAt: "2024-05-30T14:19:17.851Z",
        status: "SUCCESS",
        duration: 20635,
        __typename: "Job",
      },
      {
        id: "pz5xq17z4ba1n7isl6de2031",
        name: "build",
        createdAt: "2024-05-30T14:19:17.851Z",
        status: "SUCCESS",
        duration: 18890,
        __typename: "Job",
      },
    ],
    message: null,
    name: "Run #18",
    title: "[plugin] update version",
    cursor: "IcHZlXgiwAAA",
    status: "SUCCESS",
    __typename: "Run",
  },
  {
    id: "amzu16be4ewpjd0a9znyuotj",
    branch: "main",
    commit: "64326a0",
    date: "2024-05-30T14:18:16.439Z",
    project: "admiring-meninsky",
    projectId: "ccogwb1oau3y9oav8zwbg9zk",
    duration: 55607,
    jobs: [
      {
        id: "moog6z4srrcu3fwjx2uay5ln",
        name: "test",
        createdAt: "2024-05-30T14:18:16.425Z",
        status: "SUCCESS",
        duration: 23039,
        __typename: "Job",
      },
      {
        id: "g6t8p941shlglf3p13qcwmxu",
        name: "build",
        createdAt: "2024-05-30T14:18:16.425Z",
        status: "SUCCESS",
        duration: 32169,
        __typename: "Job",
      },
    ],
    message: null,
    name: "Run #17",
    title: "[plugin] update version",
    cursor: "IcHZlXgiwAAA",
    status: "SUCCESS",
    __typename: "Run",
  },
  {
    id: "x4e4om30gsrlxh1m3ky21u1x",
    branch: "main",
    commit: "64326a0",
    date: "2024-05-30T14:17:27.947Z",
    project: "admiring-meninsky",
    projectId: "ccogwb1oau3y9oav8zwbg9zk",
    duration: 42322,
    jobs: [
      {
        id: "lwmvsfu72epflue66uo9zopn",
        name: "test",
        createdAt: "2024-05-30T14:17:27.933Z",
        status: "SUCCESS",
        duration: 21664,
        __typename: "Job",
      },
      {
        id: "s1nhlyb29qx83xkr6p5mg1c7",
        name: "build",
        createdAt: "2024-05-30T14:17:27.933Z",
        status: "SUCCESS",
        duration: 20347,
        __typename: "Job",
      },
    ],
    message: null,
    name: "Run #16",
    title: "[plugin] update version",
    cursor: "IcHZlXgiwAAA",
    status: "SUCCESS",
    __typename: "Run",
  },
  {
    id: "kzhijmkhcwr6lx5h2jnd11zr",
    branch: "main",
    commit: "64326a0",
    date: "2024-05-30T14:15:39.535Z",
    project: "admiring-meninsky",
    projectId: "ccogwb1oau3y9oav8zwbg9zk",
    duration: 50528,
    jobs: [
      {
        id: "gsq0y27j78lt2lhnw9vgmbg2",
        name: "test",
        createdAt: "2024-05-30T14:15:39.521Z",
        status: "SUCCESS",
        duration: 32533,
        __typename: "Job",
      },
      {
        id: "k5a5vdlvjdgutdwotydt4jk9",
        name: "build",
        createdAt: "2024-05-30T14:15:39.521Z",
        status: "SUCCESS",
        duration: 17566,
        __typename: "Job",
      },
    ],
    message: null,
    name: "Run #15",
    title: "[plugin] update version",
    cursor: "IcHZlXgiwAAA",
    status: "SUCCESS",
    __typename: "Run",
  },
  {
    id: "at65tgplme2l69zfe6n7ud64",
    branch: "main",
    commit: "64326a0",
    date: "2024-05-30T14:14:35.948Z",
    project: "admiring-meninsky",
    projectId: "ccogwb1oau3y9oav8zwbg9zk",
    duration: 23565,
    jobs: [
      {
        id: "dhzrqi7qjni4tbn88a9eo89h",
        name: "test",
        createdAt: "2024-05-30T14:14:35.934Z",
        status: "SUCCESS",
        duration: 16636,
        __typename: "Job",
      },
      {
        id: "qeilrmh0t10hp5nycxxb4upz",
        name: "build",
        createdAt: "2024-05-30T14:14:35.934Z",
        status: "SUCCESS",
        duration: 6701,
        __typename: "Job",
      },
    ],
    message: null,
    name: "Run #14",
    title: "[plugin] update version",
    cursor: "IcHZlXgiwAAA",
    status: "SUCCESS",
    __typename: "Run",
  },
  {
    id: "ku7usad1ep81ow38hy24wz73",
    branch: "main",
    commit: "64326a0",
    date: "2024-05-30T14:13:00.412Z",
    project: "admiring-meninsky",
    projectId: "ccogwb1oau3y9oav8zwbg9zk",
    duration: 34670,
    jobs: [
      {
        id: "ntzd8zzu2oxu39rkah4akqqw",
        name: "test",
        createdAt: "2024-05-30T14:13:00.404Z",
        status: "SUCCESS",
        duration: 16988,
        __typename: "Job",
      },
      {
        id: "nrwm0axjzbymtru46a53yl8o",
        name: "build",
        createdAt: "2024-05-30T14:13:00.405Z",
        status: "SUCCESS",
        duration: 17399,
        __typename: "Job",
      },
    ],
    message: null,
    name: "Run #13",
    title: "[plugin] update version",
    cursor: "IcHZlXgiwAAA",
    status: "SUCCESS",
    __typename: "Run",
  },
  {
    id: "hx096gahj4t9woffp6tcwdim",
    branch: "main",
    commit: "64326a0",
    date: "2024-05-30T14:12:05.348Z",
    project: "admiring-meninsky",
    projectId: "ccogwb1oau3y9oav8zwbg9zk",
    duration: 46296,
    jobs: [
      {
        id: "hug3tcftru030qi6r3sztdjj",
        name: "test",
        createdAt: "2024-05-30T14:12:05.340Z",
        status: "SUCCESS",
        duration: 26562,
        __typename: "Job",
      },
      {
        id: "br8obw3kw2l7w1c5ruucc5ff",
        name: "build",
        createdAt: "2024-05-30T14:12:05.341Z",
        status: "SUCCESS",
        duration: 19454,
        __typename: "Job",
      },
    ],
    message: null,
    name: "Run #12",
    title: "[plugin] update version",
    cursor: "IcHZlXgiwAAA",
    status: "SUCCESS",
    __typename: "Run",
  },
  {
    id: "ld7ihjoovu8a65v75w6edrh3",
    branch: "main",
    commit: "64326a0",
    date: "2024-05-30T14:10:03.192Z",
    project: "admiring-meninsky",
    projectId: "ccogwb1oau3y9oav8zwbg9zk",
    duration: 70604,
    jobs: [
      {
        id: "bpy78tg6gqwkh2paj8ndlsqo",
        name: "test",
        createdAt: "2024-05-30T14:10:03.174Z",
        status: "SUCCESS",
        duration: 48982,
        __typename: "Job",
      },
      {
        id: "j004pg9xb7xla863okdhn1mo",
        name: "build",
        createdAt: "2024-05-30T14:10:03.174Z",
        status: "SUCCESS",
        duration: 21308,
        __typename: "Job",
      },
    ],
    message: null,
    name: "Run #11",
    title: "[plugin] update version",
    cursor: "IcHZlXgiwAAA",
    status: "SUCCESS",
    __typename: "Run",
  },
  {
    id: "m4hynmr7rjm30n8ycq600fbh",
    branch: "main",
    commit: "64326a0",
    date: "2024-05-30T14:08:32.771Z",
    project: "admiring-meninsky",
    projectId: "ccogwb1oau3y9oav8zwbg9zk",
    duration: 43102,
    jobs: [
      {
        id: "tu8iey7rtzua4b05w3z7g32v",
        name: "test",
        createdAt: "2024-05-30T14:08:32.757Z",
        status: "SUCCESS",
        duration: 20721,
        __typename: "Job",
      },
      {
        id: "lncvyej458t6lsa7en4kpguz",
        name: "build",
        createdAt: "2024-05-30T14:08:32.757Z",
        status: "SUCCESS",
        duration: 21931,
        __typename: "Job",
      },
    ],
    message: null,
    name: "Run #10",
    title: "[plugin] update version",
    cursor: "IcHZlXgiwAAA",
    status: "SUCCESS",
    __typename: "Run",
  },
  {
    id: "rxi3t2u71belkif6qsi0f2l0",
    branch: "main",
    commit: "64326a0",
    date: "2024-05-30T14:06:47.122Z",
    project: "admiring-meninsky",
    projectId: "ccogwb1oau3y9oav8zwbg9zk",
    duration: 39296,
    jobs: [
      {
        id: "ruqzw3jg76ur2sg70jpj2tbc",
        name: "test",
        createdAt: "2024-05-30T14:06:47.049Z",
        status: "SUCCESS",
        duration: 26299,
        __typename: "Job",
      },
      {
        id: "nickylgy2zx63o4glj810uaq",
        name: "build",
        createdAt: "2024-05-30T14:06:47.049Z",
        status: "SUCCESS",
        duration: 12549,
        __typename: "Job",
      },
    ],
    message: null,
    name: "Run #9",
    title: "[plugin] update version",
    cursor: "IcHZlXgiwAAA",
    status: "SUCCESS",
    __typename: "Run",
  },
  {
    id: "rxtfxrc8m17tk8oqmpbno5ew",
    branch: "main",
    commit: "64326a0",
    date: "2024-05-30T14:05:40.260Z",
    project: "admiring-meninsky",
    projectId: "ccogwb1oau3y9oav8zwbg9zk",
    duration: 83047,
    jobs: [
      {
        id: "qwadak7vto8n738rteszzat8",
        name: "test",
        createdAt: "2024-05-30T14:05:40.244Z",
        status: "SUCCESS",
        duration: 54105,
        __typename: "Job",
      },
      {
        id: "zwqrig2ih005i57x5uag6nof",
        name: "build",
        createdAt: "2024-05-30T14:05:40.245Z",
        status: "SUCCESS",
        duration: 28416,
        __typename: "Job",
      },
    ],
    message: null,
    name: "Run #8",
    title: "[plugin] update version",
    cursor: "IcHZlXgiwAAA",
    status: "SUCCESS",
    __typename: "Run",
  },
  {
    id: "xaojk56w04i7eawug9q4yoai",
    branch: "main",
    commit: "64326a0",
    date: "2024-05-30T14:04:17.514Z",
    project: "admiring-meninsky",
    projectId: "ccogwb1oau3y9oav8zwbg9zk",
    duration: 73309,
    jobs: [
      {
        id: "h2ju8fx6bo3e3bfvmxd1hszt",
        name: "test",
        createdAt: "2024-05-30T14:04:17.500Z",
        status: "SUCCESS",
        duration: 46859,
        __typename: "Job",
      },
      {
        id: "s03i79rajqisoacdjnf1wx9s",
        name: "build",
        createdAt: "2024-05-30T14:04:17.500Z",
        status: "SUCCESS",
        duration: 26013,
        __typename: "Job",
      },
    ],
    message: null,
    name: "Run #7",
    title: "[plugin] update version",
    cursor: "IcHZlXgiwAAA",
    status: "SUCCESS",
    __typename: "Run",
  },
  {
    id: "qljan7aqkbygb52xol5vjpp8",
    branch: "main",
    commit: "64326a0",
    date: "2024-05-30T14:03:21.397Z",
    project: "admiring-meninsky",
    projectId: "ccogwb1oau3y9oav8zwbg9zk",
    duration: 44673,
    jobs: [
      {
        id: "v2sct9q5ly2gcfokhjklfxpd",
        name: "test",
        createdAt: "2024-05-30T14:03:21.382Z",
        status: "SUCCESS",
        duration: 25699,
        __typename: "Job",
      },
      {
        id: "xvaat87df5z1qzw7865zkj4v",
        name: "build",
        createdAt: "2024-05-30T14:03:21.382Z",
        status: "SUCCESS",
        duration: 18621,
        __typename: "Job",
      },
    ],
    message: null,
    name: "Run #6",
    title: "[plugin] update version",
    cursor: "IcHZlXgiwAAA",
    status: "SUCCESS",
    __typename: "Run",
  },
  {
    id: "rn7geppeqb3c294qwwcn8wnn",
    branch: "main",
    commit: "64326a0",
    date: "2024-05-30T14:02:02.521Z",
    project: "admiring-meninsky",
    projectId: "ccogwb1oau3y9oav8zwbg9zk",
    duration: 51136,
    jobs: [
      {
        id: "z14dc7y19l2dnp3i3zwig6xt",
        name: "test",
        createdAt: "2024-05-30T14:02:02.515Z",
        status: "SUCCESS",
        duration: 29452,
        __typename: "Job",
      },
      {
        id: "ctz2elloaqw9ll17v7zgj18t",
        name: "build",
        createdAt: "2024-05-30T14:02:02.515Z",
        status: "SUCCESS",
        duration: 20975,
        __typename: "Job",
      },
    ],
    message: null,
    name: "Run #5",
    title: "[plugin] update version",
    cursor: "IcHZlXgiwAAA",
    status: "SUCCESS",
    __typename: "Run",
  },
  {
    id: "cqr74vyuxmlxb7yerhfcgv14",
    branch: "main",
    commit: "64326a0",
    date: "2024-05-30T14:00:06.821Z",
    project: "admiring-meninsky",
    projectId: "ccogwb1oau3y9oav8zwbg9zk",
    duration: 63649,
    jobs: [
      {
        id: "p87s1b1xlt7wykvtvgvecow0",
        name: "test",
        createdAt: "2024-05-30T14:00:06.805Z",
        status: "SUCCESS",
        duration: 34409,
        __typename: "Job",
      },
      {
        id: "b7decu7kigqarxpeh0oru4f3",
        name: "build",
        createdAt: "2024-05-30T14:00:06.805Z",
        status: "SUCCESS",
        duration: 28703,
        __typename: "Job",
      },
    ],
    message: null,
    name: "Run #4",
    title: "[plugin] update version",
    cursor: "IcHZlXgiwAAA",
    status: "SUCCESS",
    __typename: "Run",
  },
  {
    id: "emo4b7cvjvldxzigd357qi41",
    branch: "main",
    commit: "64326a0",
    date: "2024-05-30T13:58:34.972Z",
    project: "admiring-meninsky",
    projectId: "ccogwb1oau3y9oav8zwbg9zk",
    duration: 82346,
    jobs: [
      {
        id: "drkgjqc4l43fwzj6fldy0a5p",
        name: "test",
        createdAt: "2024-05-30T13:58:34.964Z",
        status: "SUCCESS",
        duration: 48136,
        __typename: "Job",
      },
      {
        id: "ec6zixy9biisx54z7h64bfz0",
        name: "build",
        createdAt: "2024-05-30T13:58:34.964Z",
        status: "SUCCESS",
        duration: 33622,
        __typename: "Job",
      },
    ],
    message: null,
    name: "Run #3",
    title: "[plugin] update version",
    cursor: "IcHZlXgiwAAA",
    status: "SUCCESS",
    __typename: "Run",
  },
  {
    id: "qj019uwyx55v8z74yp36dsxn",
    branch: "main",
    commit: "64326a0",
    date: "2024-05-28T15:32:19.489Z",
    project: "admiring-meninsky",
    projectId: "ccogwb1oau3y9oav8zwbg9zk",
    duration: 17420,
    jobs: [
      {
        id: "ik49e580m6vpr8qi4c6x94mj",
        name: "test",
        createdAt: "2024-05-28T15:32:19.489Z",
        status: "SUCCESS",
        duration: 7986,
        __typename: "Job",
      },
      {
        id: "jcf212n7xbp7ckzv42vkm008",
        name: "build",
        createdAt: "2024-05-28T15:32:19.489Z",
        status: "SUCCESS",
        duration: 9206,
        __typename: "Job",
      },
    ],
    message: null,
    name: "Run #2",
    title: "[plugin] update version",
    cursor: "IcHZlXgiwAAA",
    status: "SUCCESS",
    __typename: "Run",
  },
  {
    id: "e06fkimhve7zj2917movmvsl",
    branch: "main",
    commit: "64326a0",
    date: "2024-05-28T13:47:55.710Z",
    project: "admiring-meninsky",
    projectId: "ccogwb1oau3y9oav8zwbg9zk",
    duration: 53745,
    jobs: [
      {
        id: "r05yucwc1s2kfkyt3vlo4oy7",
        name: "test",
        createdAt: "2024-05-28T13:47:55.710Z",
        status: "SUCCESS",
        duration: 19386,
        __typename: "Job",
      },
      {
        id: "e9y0s2p15nzat75jt0518rc0",
        name: "build",
        createdAt: "2024-05-28T13:47:55.710Z",
        status: "SUCCESS",
        duration: 33873,
        __typename: "Job",
      },
    ],
    message: null,
    name: "Run #1",
    title: "[plugin] update version",
    cursor: "IcHZlXgiwAAA",
    status: "SUCCESS",
    __typename: "Run",
  },
];
