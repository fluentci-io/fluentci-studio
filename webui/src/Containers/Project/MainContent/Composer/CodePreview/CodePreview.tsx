import { FC, useEffect } from "react";
import Prism from "prismjs";
import "prismjs/components/prism-yaml";
import "prismjs/themes/prism-funky.css";
import PlateformSelect from "./PlateformSelect";
import Menu from "./Menu";

export type CodePreviewProps = {
  code: string;
};

const CodePreview: FC<CodePreviewProps> = ({ code }) => {
  useEffect(() => {
    Prism.highlightAll();
  }, [code]);

  return (
    <>
      <PlateformSelect />
      <div className="Code" style={{ marginTop: 50 }}>
        <pre style={{ position: "relative" }}>
          <Menu />
          <code className="language-yaml">{code}</code>
        </pre>
      </div>
    </>
  );
};

export default CodePreview;
