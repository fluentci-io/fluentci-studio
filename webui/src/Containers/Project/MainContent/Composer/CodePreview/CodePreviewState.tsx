import { atom } from "recoil";
import { code } from "./mocks";

export const CodePreviewState = atom<string>({
  key: "CodePreviewState",
  default: code,
});
