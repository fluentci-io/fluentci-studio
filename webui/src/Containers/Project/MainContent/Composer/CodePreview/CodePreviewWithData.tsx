import { FC, useEffect } from "react";
import CodePreview from "./CodePreview";
import { useRecoilState, useRecoilValue } from "recoil";
import { CodePreviewState } from "./CodePreviewState";
import { PlateformSelectState } from "./PlateformSelect/PlateformSelectState";
import { useParams } from "react-router-dom";
import { useExportActionsLazyQuery } from "../../../../../Hooks/GraphQL";
import { ComposerState } from "../ComposerState";

const CodePreviewWithData: FC = () => {
  const { id } = useParams();
  const plateform = useRecoilValue(PlateformSelectState);
  const actions = useRecoilValue(ComposerState);
  const [code, setCode] = useRecoilState(CodePreviewState);
  const [exportActions] = useExportActionsLazyQuery({
    variables: {
      projectId: id!,
      plateform: plateform[0].id,
    },
    fetchPolicy: "network-only",
  });

  useEffect(() => {
    exportActions({
      variables: {
        projectId: id!,
        plateform: plateform[0].id,
      },
      fetchPolicy: "network-only",
    }).then(({ data }) => {
      setCode(data?.exportActions || "");
    });
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [plateform, actions]);

  return <CodePreview code={code} />;
};

export default CodePreviewWithData;
