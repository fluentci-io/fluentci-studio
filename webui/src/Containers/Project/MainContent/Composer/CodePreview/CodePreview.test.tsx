import { render } from "@testing-library/react";
import CodePreview from "./CodePreview";
import { code } from "./mocks";
import { RecoilRoot } from "recoil";

describe("CodePreview", () => {
  it("should render the component", () => {
    const { container } = render(<CodePreview code={code} />, {
      wrapper: RecoilRoot,
    });
    expect(container).toMatchSnapshot();
  });
});
