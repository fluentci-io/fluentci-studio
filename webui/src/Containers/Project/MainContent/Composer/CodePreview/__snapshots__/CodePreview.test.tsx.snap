// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CodePreview > should render the component 1`] = `
<div>
  <div
    class="css-38orqy"
  >
    <div
      class=""
      data-baseweb="select"
    >
      <div
        class=""
      >
        <div
          class=""
        >
          <div
            class=""
            value="github"
          >
            Github Actions
          </div>
          <div
            class=""
          >
            <input
              aria-autocomplete="list"
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-label="Selected Github Actions. "
              class=""
              role="combobox"
              tabindex="0"
              value=""
            />
            <div
              class=""
            />
          </div>
        </div>
        <div
          class=""
        >
          <svg
            class=""
            data-baseweb="icon"
            title="open"
            viewBox="0 0 24 24"
          >
            <title>
              open
            </title>
            <path
              d="M12.7071 15.2929L17.1464 10.8536C17.4614 10.5386 17.2383 10 16.7929 10L7.20711 10C6.76165 10 6.53857 10.5386 6.85355 10.8536L11.2929 15.2929C11.6834 15.6834 12.3166 15.6834 12.7071 15.2929Z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
  <div
    class="Code"
    style="margin-top: 50px;"
  >
    <pre
      class="language-yaml"
      style="position: relative;"
      tabindex="0"
    >
      <div
        class="css-868fxl"
      >
        <button
          class="css-19nl8qs"
        >
          <svg
            aria-hidden="true"
            class="StyledIconBase-sc-ea9ulj-0 gEpWWb"
            fill="currentColor"
            focusable="false"
            height="18"
            viewBox="0 0 512 512"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="336"
              rx="57"
              ry="57"
              stroke="currentColor"
              stroke-linejoin="round"
              stroke-width="32"
              width="336"
              x="128"
              y="128"
            />
            <path
              d="m383.5 128 .5-24a56.16 56.16 0 0 0-56-56H112a64.19 64.19 0 0 0-64 64v216a56.16 56.16 0 0 0 56 56h24"
              fill="none"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="32"
            />
          </svg>
        </button>
        <button
          class="css-19nl8qs"
        >
          <svg
            aria-hidden="true"
            class="StyledIconBase-sc-ea9ulj-0 gEpWWb"
            fill="currentColor"
            focusable="false"
            height="18"
            viewBox="0 0 16 16"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M.5 9.9a.5.5 0 0 1 .5.5v2.5a1 1 0 0 0 1 1h12a1 1 0 0 0 1-1v-2.5a.5.5 0 0 1 1 0v2.5a2 2 0 0 1-2 2H2a2 2 0 0 1-2-2v-2.5a.5.5 0 0 1 .5-.5z"
            />
            <path
              d="M7.646 11.854a.5.5 0 0 0 .708 0l3-3a.5.5 0 0 0-.708-.708L8.5 10.293V1.5a.5.5 0 0 0-1 0v8.793L5.354 8.146a.5.5 0 1 0-.708.708l3 3z"
            />
          </svg>
        </button>
      </div>
      <code
        class="language-yaml"
      >
        <span
          class="token key atrule"
        >
          name
        </span>
        <span
          class="token punctuation"
        >
          :
        </span>
         lint and test

        <span
          class="token key atrule"
        >
          on
        </span>
        <span
          class="token punctuation"
        >
          :
        </span>
        
  
        <span
          class="token key atrule"
        >
          push
        </span>
        <span
          class="token punctuation"
        >
          :
        </span>
        
    
        <span
          class="token key atrule"
        >
          branches
        </span>
        <span
          class="token punctuation"
        >
          :
        </span>
        
      
        <span
          class="token punctuation"
        >
          -
        </span>
         main

        <span
          class="token key atrule"
        >
          jobs
        </span>
        <span
          class="token punctuation"
        >
          :
        </span>
        
  
        <span
          class="token key atrule"
        >
          tests
        </span>
        <span
          class="token punctuation"
        >
          :
        </span>
        
    
        <span
          class="token key atrule"
        >
          runs-on
        </span>
        <span
          class="token punctuation"
        >
          :
        </span>
         ubuntu
        <span
          class="token punctuation"
        >
          -
        </span>
        latest
    
        <span
          class="token key atrule"
        >
          steps
        </span>
        <span
          class="token punctuation"
        >
          :
        </span>
        
      
        <span
          class="token punctuation"
        >
          -
        </span>
         
        <span
          class="token key atrule"
        >
          uses
        </span>
        <span
          class="token punctuation"
        >
          :
        </span>
         actions/checkout@v2
      
        <span
          class="token punctuation"
        >
          -
        </span>
         
        <span
          class="token key atrule"
        >
          name
        </span>
        <span
          class="token punctuation"
        >
          :
        </span>
         Setup Fluent CI
        
        <span
          class="token key atrule"
        >
          uses
        </span>
        <span
          class="token punctuation"
        >
          :
        </span>
         fluentci
        <span
          class="token punctuation"
        >
          -
        </span>
        io/setup
        <span
          class="token punctuation"
        >
          -
        </span>
        fluentci@v5
        
        <span
          class="token key atrule"
        >
          with
        </span>
        <span
          class="token punctuation"
        >
          :
        </span>
        
          
        <span
          class="token key atrule"
        >
          wasm
        </span>
        <span
          class="token punctuation"
        >
          :
        </span>
         
        <span
          class="token boolean important"
        >
          true
        </span>
        
          
        <span
          class="token key atrule"
        >
          pipeline
        </span>
        <span
          class="token punctuation"
        >
          :
        </span>
         deno
          
        <span
          class="token key atrule"
        >
          args
        </span>
        <span
          class="token punctuation"
        >
          :
        </span>
         
        <span
          class="token punctuation"
        >
          |
        </span>
        <span
          class="token scalar string"
        >
          
            fmt
            lint
            test
            coverage
        </span>
        
      
        <span
          class="token punctuation"
        >
          -
        </span>
         
        <span
          class="token key atrule"
        >
          name
        </span>
        <span
          class="token punctuation"
        >
          :
        </span>
         Upload to Codecov
        
        <span
          class="token key atrule"
        >
          run
        </span>
        <span
          class="token punctuation"
        >
          :
        </span>
         fluentci run 
        <span
          class="token punctuation"
        >
          -
        </span>
        <span
          class="token punctuation"
        >
          -
        </span>
        wasm codecov upload
        
        <span
          class="token key atrule"
        >
          env
        </span>
        <span
          class="token punctuation"
        >
          :
        </span>
        
          
        <span
          class="token key atrule"
        >
          CODECOV_TOKEN
        </span>
        <span
          class="token punctuation"
        >
          :
        </span>
         $
        <span
          class="token punctuation"
        >
          {
        </span>
        <span
          class="token punctuation"
        >
          {
        </span>
         secrets.CODECOV_TOKEN 
        <span
          class="token punctuation"
        >
          }
        </span>
        <span
          class="token punctuation"
        >
          }
        </span>
      </code>
    </pre>
  </div>
</div>
`;
